core/domain/user_jobs_one_off.py

```diff
@@ -324,3 +327,53 @@ def map(user_model):
     @staticmethod
     def reduce(key, stringified_values):
         pass
+
+
+class UserRolesMigrationOneOffJob(jobs.BaseMapReduceJobManager):
+    """A one time job to attach a new field (representing role of user) to the
+    UserSettingsModel. The job will load all existing users from the datastore,
+    look for the role to assign them and then store them back thus updating the
+    table schema and adding appropiate roles to users.
+    """
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(user_model):
+        admin_usernames = config_domain.Registry.get_config_property(
+            'admin_usernames')
+        moderator_usernames = config_domain.Registry.get_config_property(
+            'moderator_usernames')
+        banned_usernames = config_domain.Registry.get_config_property(
+            'banned_usernames')
+        collection_editors = config_domain.Registry.get_config_property(
+            'collection_editor_whitelist')
+        email_senders = config_domain.Registry.get_config_property(
+            'whitelisted_email_senders')
+        admin_usernames.value.extend(email_senders.value)
```

Comment:
I don't think you should change admin_usernames.value here. Just treat them separately.